Sources/ArgumentParser/Parsable Properties/Option.swift

```diff
@@ -281,6 +281,67 @@ extension Option where Value: ExpressibleByArgument {
     })
   }
 
+  /// Creates a required property that reads its value from a labeled option.
+  ///
+  /// This initializer is used when you declare an `@Option`-attributed property
+  /// that has an `EnumerableOptionValue` type, providing a default value:
+  ///
+  /// ```swift
+  /// enum Color: String, EnumerableOptionValue {
+  ///   case red
+  ///   case blue
+  ///   case yellow
+  ///
+  ///   public var description: String {
+  ///     switch self {
+  ///       case .red:
+  ///         return "A red color."
+  ///       case .blue:
+  ///         return "A blue color."
+  ///       case .yellow:
+  ///         return "A yellow color."
+  ///     }
+  ///   }
+  /// }
+  ///
+  /// @Option var color: Color = .red
+  /// ```
+  ///
+  /// - Parameters:
+  ///   - name: A specification for what names are allowed for this option.
+  ///   - parsingStrategy: The behavior to use when looking for this option's
+  ///     value.
+  ///   - help: Information about how to use this option. If a discussion property is added, this will
+  ///     be appended to the top of the list of `EnumerableOptionValue` descriptions.
+  ///   - completion: The type of command-line completion provided for this
+  ///     option.
+  public init(
+    wrappedValue: Value,
+    name: NameSpecification = .long,
+    parsing parsingStrategy: SingleValueParsingStrategy = .next,
+    help: ArgumentHelp? = nil,
+    completion: CompletionKind? = nil
+  ) where Value: EnumerableOptionValue {
```

Comment:
For sure - I just tried this out locally by adding the following to the beginning of each `ExpressibleByArgument` init:

```swift
var help = help
if let type = Value.self as? (any EnumerableOptionValue.Type) {
  help = ArgumentHelp.init(
  help?.abstract ?? "",
  discussion: help?.discussion,
  options: type, // retaining necessary type information for enumerable values here
  valueName: help?.valueName,
  visibility: help?.visibility ?? .default
  )
}
```

This way we can still retain the modified `ArgumentHelp` for types implementing `EnumerableOptionValue` (or whatever we decide to rename it to!). 

Alternatively, I can modify the `ArgumentHelp` init to take the `options` parameter and have it be optional, and can remove some code so it comes down to this:

```swift
// ArgumentHelp init:
public init(
  _ abstract: String = "",
  discussion: String? = nil,
  valueName: String? = nil,
  options: (any EnumerableOptionValue.Type)? = nil, // made this optional
  visibility: ArgumentVisibility = .default)
{
  self.abstract = abstract
  self.discussion = discussion
  self.valueName = valueName
  self.options = options
  self.visibility = visibility
} 
```

and the resulting `ExpressibleByArgument` inits would have some additional code that looks something like this:

```swift
self.init(_parsedValue: .init { key in
  let arg = ArgumentDefinition(
    container: Bare<Value>.self,
    key: key,
    kind: .name(key: key, specification: name),
    help: .init(
      help?.abstract ?? "",
      discussion: help?.discussion,
      valueName: help?.valueName,
      options: Value.self as? (any EnumerableOptionValue.Type),
      visibility: help?.visibility ?? .default
    ),
    parsingStrategy: parsingStrategy.base,
    initial: wrappedValue,
    completion: completion)

  return ArgumentSet(arg)
})
```

If this seems good to you, I can go ahead and change it!

Reply:
Sorry for the delay here! I've been wrestling with whether we can avoid the `EnumerableOptionValue` protocol altogether, since it seems to be a refinement of the `ExpressibleByArgument` protocol that we're already handling for a couple slightly different uses. In particular, adding features like this one to the library is challenging in that we need to design a source-compatible addition that makes some kind of intuitive sense with the features that are already present.

The most related existing feature supported by the `ExpressibleByArgument` protocol is the ability to automatically include value names in the help description when a type is `CaseIterable`. Since the change in this PR kind of augments that to include descriptions as well as value names, I think we can piggy back on that existing design.

In particular, what would you think about adding another customization point to `ExpressibleByArgument`:

```swift
public protocol ExpressibleByArgument {
  // ...other members...
  
  // Existing static var for value strings:

  /// An array of all possible strings that can convert to a value of this
  /// type, for display in the help screen.
  ///
  /// The default implementation of this property returns an empty array. If the
  /// conforming type is also `CaseIterable`, the default implementation returns
  /// an array with a value for each case.
  static var allValueStrings: [String] { get }

  // New static var for value descriptions:
  
  static var allValueDescriptions: [String: String] { get }
}
```

We could provide a default implementation that returns an empty dictionary, and then any type that wants to provide a more expanded help screen could implement `allValueDescriptions` to provide corresponding descriptions for each element of the `allValueStrings` array. Internally, we can use similar machinery to capture those descriptions for use when displaying the help screen or generating the JSON help description.

Reply:
This is a great idea! Thanks for the feedback - I tried to implement this locally and it seems to work well.